File: map_python/src/Restaurant/restaurant_map.py
from dataclasses import dataclass, field

from .calorie import CalorieValue
from .ingredient import Ingredient
from .prepared_product import PreparedProduct
from .stock_item import StockItem
from .stock_loss import StockLoss


@dataclass(slots=True)
class RestaurantMap:
    raw_sources: list[Ingredient] = field(default_factory=list)
    stocks: list[StockItem] = field(default_factory=list)
    prepared_products: list[PreparedProduct] = field(default_factory=list)
    preparation_losses: list[StockLoss] = field(default_factory=list)
    table_reset_losses: list[StockLoss] = field(default_factory=list)
    calories: list[CalorieValue] = field(default_factory=list)

File: map_ts/src/Restaurant/restaurant-map.ts
import type { Ingredient } from "./ingredient.js";
import type { StockItem } from "./stock-item.js";
import type { PreparedProduct } from "./prepared-product.js";
import type { StockLoss } from "./stock-loss.js";
import type { CalorieValue } from "./calorie.js";

export type RestaurantMap = {
  rawSources: Ingredient[];

  stocks: StockItem[];

  preparedProducts: PreparedProduct[];

  preparationLosses: StockLoss[];

  tableResetLosses: StockLoss[];

  calories: CalorieValue[];
};